### PR TITLE
FEM-1665

### PR DESCRIPTION
**Pull request by giladna**
Add query strings for download playmanifest request:

playbackType | offline
clientTag | playkit-dtg/android-2.2.3
referrer | TXlBcHA=
playSessionId | 984d06f9-0e65-44e9-891e-3ab825a69f97

### CODE REVIEW

File: dtglib/src/main/java/com/kaltura/dtg/clear/ContentManagerImp.java [hunk 1]
```diff
@@ -307,5 +316,21 @@ public Settings getSettings() {
     public void setAutoResumeItemsInProgress(boolean autoStartItemsInProgress) {
         this.autoResumeItemsInProgress = autoStartItemsInProgress;
     }
+
+    public String getSessionId() {
+        return sessionId;
+    }
+
+    public void setSessionId(String sessionId) {
```
Comment (noamtamim) [inline dtglib/src/main/java/com/kaltura/dtg/clear/ContentManagerImp.java]: No public setter. This should be a read-only property.

File: dtglib/src/main/java/com/kaltura/dtg/clear/ContentManagerImp.java [hunk 2]
```diff
@@ -307,5 +316,21 @@ public Settings getSettings() {
...
+        this.sessionId = sessionId;
+    }
+
+    public String getApplicationName() {
+        return applicationName;
+    }
+
+    public void setApplicationName(String applicationName) {
```
Comment (noamtamim) [inline dtglib/src/main/java/com/kaltura/dtg/clear/ContentManagerImp.java]: Move to settings.

File: dtglib/src/main/java/com/kaltura/dtg/clear/DefaultDownloadService.java [hunk 3]
```diff
@@ -39,8 +40,12 @@
 public class DefaultDownloadService extends Service {
 
     private static final String TAG = "DefaultDownloadService";
+    public static final String VERSION_STRING = BuildConfig.VERSION_NAME;
```
Comment (noamtamim) [inline dtglib/src/main/java/com/kaltura/dtg/clear/DefaultDownloadService.java]: Better move VERSION_STRING and CLIENT_TAG to ContentManager (the abstract class).

File: dtglib/src/main/java/com/kaltura/dtg/clear/DefaultDownloadService.java [hunk 4]
```diff
@@ -312,12 +317,12 @@ private void downloadMetadata(DefaultDownloadItem item) throws IOException {
         if (contentURL.startsWith("widevine")) {
             contentURL = contentURL.replaceFirst("widevine", "http");
         }
-        
+        Uri contentUri = Uri.parse(contentURL);
```
Comment (noamtamim) [inline dtglib/src/main/java/com/kaltura/dtg/clear/DefaultDownloadService.java]: Why is this change required here?

File: dtglib/src/main/java/com/kaltura/dtg/ContentManager.java [hunk 5]
```diff
@@ -11,8 +11,8 @@
  * Created by Noam Tamim @ Kaltura on 28/09/2016.
  */
 public abstract class ContentManager {
-    public static ContentManager getInstance(Context context) {
-        return ContentManagerImp.getInstance(context);
+    public static ContentManager getInstance(Context context, String applicationName) {
```
[the remark this answers is not in the document]
Comment (giladna) [reply dtglib/src/main/java/com/kaltura/dtg/ContentManager.java]: fixed

File: dtglib/src/main/java/com/kaltura/dtg/clear/ContentManagerImp.java [hunk 6]
```diff
@@ -77,14 +79,21 @@ public void onTracksAvailable(DownloadItem item, DownloadItem.TrackSelector trac
...
     private boolean autoResumeItemsInProgress = true;
+    private DownloadRequestParams.Adapter adapter;
     private Settings settings = new Settings();
 
-    private ContentManagerImp(Context context) {
+    private ContentManagerImp(Context context, String applicationName) {
         this.context = context.getApplicationContext();
+        this.sessionId =  UUID.randomUUID().toString();
+        this.applicationName = applicationName;
+        this.adapter = new KalturaDownloadRequestAdapter(sessionId, applicationName);
```
[the remark this answers is not in the document]
Comment (giladna) [reply dtglib/src/main/java/com/kaltura/dtg/clear/ContentManagerImp.java]: fixed

File: dtglib/src/main/java/com/kaltura/dtg/clear/ContentManagerImp.java (hunk 1 above)
Comment (giladna) [reply dtglib/src/main/java/com/kaltura/dtg/clear/ContentManagerImp.java]: fixed

File: dtglib/src/main/java/com/kaltura/dtg/clear/ContentManagerImp.java (hunk 2 above)
Comment (giladna) [reply dtglib/src/main/java/com/kaltura/dtg/clear/ContentManagerImp.java]: removed not needed

File: dtglib/src/main/java/com/kaltura/dtg/clear/DefaultDownloadService.java (hunk 3 above)
Comment (giladna) [reply dtglib/src/main/java/com/kaltura/dtg/clear/DefaultDownloadService.java]: fixed

File: dtglib/src/main/java/com/kaltura/dtg/clear/DefaultDownloadService.java (hunk 4 above)
Comment (giladna) [reply dtglib/src/main/java/com/kaltura/dtg/clear/DefaultDownloadService.java]: end with is not correct after adding query params
need to use the path sement

File: dtgapp/src/main/java/com/kaltura/dtgapp/MainActivity.java [hunk 7]
```diff
@@ -161,12 +161,12 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
 
-        final ContentManager contentManager = ContentManager.getInstance(this, "MyApp");
+        final ContentManager contentManager = ContentManager.getInstance(this);
 
         final HashMap<String, Long> downloadStartTime = new HashMap<>();
 
         contentManager.getSettings().maxConcurrentDownloads = 4;
-
+        contentManager.getSettings().applicationName = "JAVA";
```
Comment (noamtamim) [inline dtgapp/src/main/java/com/kaltura/dtgapp/MainActivity.java]: As sample code, a better sample string would be what you used before -- "MyApp" or similar.
